service/resource/persistentvolume/state_test.go

```diff
@@ -0,0 +1,142 @@
+package persistentvolume
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/micrologger/microloggertest"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func Test_Resource_RecyclePersistentVolume_GetCurrentState(t *testing.T) {
+	testCases := []struct {
+		Obj                             interface{}
```

Comment:
No need to have this fields exported (`obj` instead of `Obj` works). It would also be nice to have a `description` string field with an explanation of the test.